Library/Homebrew/cmd/--cellar.rb

```diff
@@ -25,7 +25,7 @@ def __cellar
     if Homebrew.args.named.blank?
       puts HOMEBREW_CELLAR
     else
-      puts ARGV.resolved_formulae.map(&:rack)
```

Comment:
Can the `ARGV` method be `odeprecated`?

Reply:
Yes. Updating the method accordingly

Reply:
Great, thanks. If that can be done for any of the other methods you've done in previous PRs that would be great, too.

Reply:
The way we run `postinstall` is a bit tricky and is blocking `ARGV#resolved_formulae` from being deprecated. 
1. [`cmd/postinstall.rb`](https://github.com/Homebrew/brew/blob/master/Library/Homebrew/cmd/postinstall.rb) calls [`FormulaInstaller#post_install`](https://github.com/Homebrew/brew/blob/master/Library/Homebrew/formula_installer.rb#L909) which then runs [`Homebrew/postinstall.rb`](https://github.com/Homebrew/brew/blob/master/Library/Homebrew/postinstall.rb) in a sandbox

2. `Homebrew/postinstall.rb` does not use the parser but uses `ARGV.resolved_formulae` 

We could either
 - use `cli/parser` to parse args within `Homebrew/postinstall.rb`
 - Add fallback to ARGV in `CLI::Args#resolved_formulae`

@MikeMcQuaid Need your input on the above two approaches. I can attempt this in a separate PR.

Reply:
> * use `cli/parser` to parse args within `Homebrew/postinstall.rb`

This seems better 👍. Thanks!

Are there any other `ARGV` methods that can be removed or `odeprecated` that were previously ported?

Reply:
> Are there any other `ARGV` methods that can be removed or `odeprecated` that were previously ported?

Checked these and done them!